### PR TITLE
Mild refactor of time sources processing

### PR DESCRIPTION
**Pull request by fernflower**
In case includefile directive was specified and
in an included config servers with the same name
were present this resulted in a duplicate pool servers
in chrony.conf.
Besides this patch addresses the issue with wrong
numbers of pool servers required for conversion
(NOT IMPLEMENTED YET, waiting for confirmation from mlichvar)

Some minor code refactor has also been done.

### CODE REVIEW

**Comment by mlichvar** [comment]
I think the code works as expected. If there are duplicates in the original ntp configuration, they should preserved in the generated chrony configuration. The pool directive uses four servers, so only a set of four servers should be converted to pool.

File: ntp2chrony/ntp2chrony.py [hunk 1]
```diff
@@ -356,44 +356,29 @@ def get_processed_time_sources(self):
...
-        processed_sources = []
-        for source in self.time_sources:
-            if source["type"] == "server":
-                if source["address"] in remove_servers:
-                    continue
-                if source["address"] in convert_servers:
-                    source["type"] = "pool"
-            processed_sources.append(source)
+            # verify that parameters are the same for all servers in the pool
+            if not all([p[1]['options'] == pool[0][1]['options'] for p in pool]):
```
Comment (mlichvar) [inline ntp2chrony/ntp2chrony.py]: Please use double quotes as the rest of the code.

Comment (fernflower) [reply ntp2chrony/ntp2chrony.py]: Done, sorry I missed that one.

File: ntp2chrony/ntp2chrony.py [hunk 2]
```diff
@@ -356,44 +356,29 @@ def get_processed_time_sources(self):
...
-                    source["type"] = "pool"
-            processed_sources.append(source)
+            # verify that parameters are the same for all servers in the pool
+            if not all([p[1]["options"] == pool[0][1]["options"] for p in pool]):
+                break
+            remove_servers.update([pool[i][1]["address"] for i in [0, 1, 3]])
+            pool[2][1]["type"] = "pool"
 
-        return processed_sources
+        return [s for s in self.time_sources if s["address"] not in remove_servers]
```
Comment (mlichvar) [inline ntp2chrony/ntp2chrony.py]: I think this should remove only server sources, not peers or pools.

Comment (fernflower) [reply ntp2chrony/ntp2chrony.py]: Sorry not getting you, why are you saying that peers or pools are removed? remove_servers contains a set of server sources 0, 1, 3 only.
The final list comprehension is equivalent to the original code
```
        processed_sources = []
        for source in self.time_sources:
            if source["type"] == "server":
                if source["address"] in remove_servers:
                    continue
                if source["address"] in convert_servers:
                    source["type"] = "pool"
            processed_sources.append(source)
        return processed_sources
```

Comment (mlichvar) [reply ntp2chrony/ntp2chrony.py]: There is no check of the type. It's not equivalent. You can try it with this config:
```
server 0.pool.ntp.org
server 1.pool.ntp.org
server 2.pool.ntp.org
server 3.pool.ntp.org
peer 0.pool.ntp.org
```

Comment (fernflower) [reply ntp2chrony/ntp2chrony.py]: Understood, thanks, added this as another test case to fixate.

**Comment by mlichvar** [comment]
Applied. Thanks.

**Pull request closed without merging**